packages/material-ui/src/TablePagination/TablePagination.d.ts

```diff
@@ -24,8 +24,18 @@ export interface TablePaginationTypeMap<P, D extends React.ElementType> {
       labelRowsPerPage?: React.ReactNode;
       nextIconButtonProps?: Partial<IconButtonProps>;
       nextIconButtonText?: string;
-      onChangePage: (event: React.MouseEvent<HTMLButtonElement> | null, page: number) => void;
+      /**
+       * Deprecated. Will be removed in v5. Please use the onPageChange prop instead.
+       * @deprecated
+       */
```

Comment:
What do you think about this approach instead, keeping the previous description of the prop but add a deprecation with a direct upgrade path?

```suggestion
      /**
       * Callback fired when the page is changed.
       *
       * @param {object} event The event source of the callback.
       * @param {number} page The page selected.
       * @deprecated Use the onPageChange prop instead
       */
```

Reply:
Done